### PR TITLE
Rule: Expression with label

### PR DESCRIPTION
**Pull request by tagantroy**
This PR implemented #173

### CODE REVIEW

**Comment by arturbosch** [comment]
Same formatting + documentation + default config + complexity provider things as in #174.

Thank you for this PR!

File: detekt-cli/src/main/resources/default-detekt-config.yml [hunk 1]
```diff
@@ -40,6 +40,8 @@ complexity:
     threshold: 10
   ComplexCondition:
     threshold: 3
+  LabeledExpression:
```
Comment (arturbosch) [inline detekt-cli/src/main/resources/default-detekt-config.yml]: I think this rule is pretty high opinionated, we should leave it as false for some time after we have more feedback?

Comment (tagantroy) [inline detekt-cli/src/main/resources/default-detekt-config.yml]: Ok, should I keep it in complexity configuration block?

Comment (arturbosch) [inline detekt-cli/src/main/resources/default-detekt-config.yml]: I think it fits the best there for now.

Comment (tagantroy) [inline detekt-cli/src/main/resources/default-detekt-config.yml]: done

**Comment by arturbosch** [comment]
Please next time take a look at [contributing](https://github.com/arturbosch/detekt/blob/master/CONTRIBUTING.md) especially at this [link](https://chris.beams.io/posts/git-commit/) to write commit messages. The commit messages for detekt should start with an imperative ^^

Big thanks for the new rules :+1:

**Comment by Mauin** [comment]
@arturbosch You could use the "Squash and Merge" feature for merging the PRs. That will squash all commits into one commit on master and name it according to the name of the PR. Might make the commit history on master a bit more readable.

**Comment by vanniktech** [comment]
In the settings of this repository, you can also turn the other options off so that you'll always use this and not the other options by accident:

screen shot 2017-07-09 at 15 55 01

**Comment by arturbosch** [comment]
I'am using the `rebase merging` option. Thx for the info. In some cases squash and merge will be useful, but in most cases I prefer a history with smaller commits :)

**Comment by vanniktech** [comment]
Squash merging will always yield you one commit on master while rebase merging can give you more depending on the number of commits in the PR.

**Pull request merged**